Fix typo in show method of OSGB36

anowacki opened the pull request:
I assume that the desired output of `Base.show(io, ::OSGB36)` is indeed 'osgb36' rather than 'osbg84', but if I'm wrong then no matter.

I'm also not sure how to test `show` methods in general, so if that is important I'd be very grateful for any guidance.

visr commented:
Thanks! Would be good to add a test as well, you can use `sprint` for this:

```julia
julia> @test sprint(show, OSGB36()) == "osgb36"
Test Failed
  Expression: sprint(show, OSGB36()) == "osgb36"
   Evaluated: "osbg84" == "osgb36"
```

anowacki commented:
I've added tests for all the `show` methods for the datums—thanks for the guidance.

I should also come clean and say that I added a `show` method for `GDA94` as I guessed that there was no particular reason for it not to have one.  However, I've no problem reverting that change if desired.

visr commented:
Yeah, well spotted, thanks for adding that as well.
